src/eureka/S5_lightcurve_fitting/s5_fit.py

```diff
@@ -410,18 +409,68 @@ def fitlc(eventlabel, ecf_path=None, s4_meta=None, input_meta=None):
                     # flux values
                     flux, flux_err = util.normalize_spectrum(meta, flux,
                                                              flux_err)
-
-                    meta, params = fit_channel(meta, time, flux, channel,
-                                               flux_err, eventlabel, params,
-                                               log, longparamlist, time_units,
-                                               paramtitles, chanrng, ld_coeffs,
-                                               xpos, ypos, xwidth, ywidth)
-
-                    # Save results
-                    log.writelog('Saving results', mute=(not meta.verbose))
-                    me.saveevent(meta, (meta.outputdir+'S5_'+meta.eventlabel +
-                                        "_Meta_Save"), save=[])
-
+                                 
+                    if hasattr(meta, 'fit_bins'):
+                        if ((channel > meta.fit_bins[0]) and 
+                           (channel < meta.fit_bins[1])):
+                            try:
+                                meta, params = fit_channel(meta, time, flux, 
+                                                           channel, flux_err, 
+                                                           eventlabel, params,
+                                                           log, longparamlist, 
+                                                           time_units, 
+                                                           paramtitles, 
+                                                           chanrng, ld_coeffs,
+                                                           xpos, ypos, 
+                                                           xwidth, ywidth)
+                            except:
```

Comment:
Except statements should only be used to catch particular expected exceptions (that's a tough phrase to read/write XD). At a bare minimum, we should still print the error message to the terminal (otherwise people will have no idea how/why their fit failed), and ideally we'd only catch exceptions that we expect to encounter. For example, the danger with how this is currently written is that hitting Ctrl+c will cause an "exception" which will then trigger this except clause and the fit will just move on to the next channel. So if I realize I forgot to change a parameter in my ECF/EPF, I have to do Ctrl+c once per channel before the fitting run finally gets killed.

Reply:
Do you have an alternative suggestion here? Because the problem arises when you set up a large run you intend to go overnight and then it crashes halfway because of a bad channel for any number of reasons.

Reply:
I'm not sure I've ever encountered such a crash before - can you give me an example of the kinds of issues you've run into before? Were they all related to some parameter not having any dynamic range when making the corner plot? Or what other kinds of sporadic crashes have you encountered?

Reply:
Hmm, Not sure I have notes on these anywhere. I'm happy to remove this for now until it's encountered again and then we can look at the underlying cause of the crash.

Reply:
My guess is the crashes are related to memory issues.  We have a server that can be running several JWST analyses at different steps and sometimes the load is too much.  Since we can't be sure what the exception will be, my suggestion is to print the exception and then pause for ~5 seconds.  This should give the user enough time to Ctrl+c again and avoid the repetitive loop mentioned by Taylor.

Reply:
My issues are rarely memory related (that typically crashes the whole thing, not just a single channel), and more like every point (or most points) in a channel are masked for some reason, or weird things with errors, or something else. Single channels crashing for random reasons happens quite often to me at native pixel res fitting. I prefer skipping and letting the rest continue, perhaps with a print out at the end to remind you which channels failed. With the option to fit only specific channels that is also in this PR that would allow the user to go back and try to refit only the failed channels, as well.